beacon-chain/state/fieldtrie/field_trie_helpers.go

```diff
@@ -178,10 +179,19 @@ func handle32ByteMVslice(mv multi_value_slice.MultiValueSliceComposite[[32]byte]
 func handleValidatorMVSlice(mv multi_value_slice.MultiValueSliceComposite[*ethpb.Validator], indices []uint64, convertAll bool) ([][32]byte, error) {
 	length := len(indices)
 	if convertAll {
-		return stateutil.OptimizedValidatorRoots(mv.Value(mv.State()))
+		vals := mv.Value(mv.State())
+		rovals := make([]validator.ReadOnlyValidator, len(vals))
+		var err error
+		for i, v := range vals {
+			rovals[i], err = validator.NewValidator(v)
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
It takes 7ms to convert 1 million validators, so this extra code is not a big issue